scr/main/java/bittercode/model/User.java

```diff
@@ -69,20 +69,22 @@ public void setRoles(List<UserRole> roles) {
         this.roles = roles;
     }
 
-//    public static User retrieveFromHttpServletRequest(HttpServletRequest req) {
-//        User user = new User();
-//        String pWord = req.getParameter(UsersDBConstants.COLUMN_PASSWORD);
-//        String fName = req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME);
-//        String lName = req.getParameter(UsersDBConstants.COLUMN_LASTNAME);
-//        String addr = req.getParameter(UsersDBConstants.COLUMN_ADDRESS);
-//        String phNo = req.getParameter(UsersDBConstants.COLUMN_PHONE);
-//        String mailId = req.getParameter(UsersDBConstants.COLUMN_MAILID);
-//        user.setEmailId(mailId);
-//        user.setFirstName(fName);
-//        user.setLastName(lName);
-//        user.setPassword(pWord);
-//        user.setPhone(Long.parseLong(phNo));
-//        user.setAddress(addr);
-//        return user;
-//    }
-}
+    public static User retrieveFromHttpServletRequest(HttpServletRequest req) {
+        User user = new User();
+        String pWord = req.getParameter(UsersDBConstants.COLUMN_PASSWORD);
+        String fName = req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME);
+        String lName = req.getParameter(UsersDBConstants.COLUMN_LASTNAME);
+        String addr = req.getParameter(UsersDBConstants.COLUMN_ADDRESS);
+        String phNo = req.getParameter(UsersDBConstants.COLUMN_PHONE);
+        String mailId = req.getParameter(UsersDBConstants.COLUMN_MAILID);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding parameter validation checks</b></div>
<div id="fix">

Consider validating the request parameters before using them. Parameters like `phNo` could be null or non-numeric which would cause `NumberFormatException`. A similar issue was also found in scr/main/java/bittercode/model/User.java (line 85).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -74,13 +74,19 @@
 -        String pWord = req.getParameter(UsersDBConstants.COLUMN_PASSWORD);
 -        String fName = req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME);
 -        String lName = req.getParameter(UsersDBConstants.COLUMN_LASTNAME);
 -        String addr = req.getParameter(UsersDBConstants.COLUMN_ADDRESS);
 -        String phNo = req.getParameter(UsersDBConstants.COLUMN_PHONE);
 -        String mailId = req.getParameter(UsersDBConstants.COLUMN_MAILID);
 +        String pWord = validateParam(req.getParameter(UsersDBConstants.COLUMN_PASSWORD));
 +        String fName = validateParam(req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME));
 +        String lName = validateParam(req.getParameter(UsersDBConstants.COLUMN_LASTNAME));
 +        String addr = validateParam(req.getParameter(UsersDBConstants.COLUMN_ADDRESS));
 +        String phNo = validateParam(req.getParameter(UsersDBConstants.COLUMN_PHONE));
 +        String mailId = validateParam(req.getParameter(UsersDBConstants.COLUMN_MAILID));
 +        
 +        if (!phNo.matches("\\d+")) {
 +            throw new IllegalArgumentException("Phone number must contain only digits");
 +        }
```


```suggestion
        String pWord = validateParam(req.getParameter(UsersDBConstants.COLUMN_PASSWORD));
        String fName = validateParam(req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME));
        String lName = validateParam(req.getParameter(UsersDBConstants.COLUMN_LASTNAME));
        String addr = validateParam(req.getParameter(UsersDBConstants.COLUMN_ADDRESS));
        String phNo = validateParam(req.getParameter(UsersDBConstants.COLUMN_PHONE));
        String mailId = validateParam(req.getParameter(UsersDBConstants.COLUMN_MAILID));
        
        if (!phNo.matches("\\d+")) {
            throw new IllegalArgumentException("Phone number must contain only digits");
        }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/8#issuecomment-2527744697>#4874c2</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged